flink/flink-1.14/src/main/java/io/tidb/bigdata/flink/connector/sink/operator/TiDBMiniBatchWriteOperator.java

```diff
@@ -0,0 +1,69 @@
+package io.tidb.bigdata.flink.connector.sink.operator;
+
+import io.tidb.bigdata.flink.connector.sink.TiDBSinkOptions;
+import io.tidb.bigdata.tidb.RowBuffer;
+import io.tidb.bigdata.tidb.TiDBEncodeHelper;
+import io.tidb.bigdata.tidb.TiDBWriteHelper;
+import io.tidb.bigdata.tidb.TiDBWriteMode;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.tikv.common.BytePairWrapper;
+import org.tikv.common.ByteWrapper;
+import org.tikv.common.meta.TiTimestamp;
+
+/**
+ * The operator for flushing rows to TiDB on mode MiniBatch.
+ * During each flush, the rows are committed to TiDB in a transaction.
+ */
+public class TiDBMiniBatchWriteOperator extends TiDBWriteOperator {
```

Comment:
I think we should implement CheckpointListener interface: 
1. Flush current rows when notifyCheckpointComplete;
2. Abandon current rows when notifyCheckpointAborted.